### PR TITLE
[pallet-xcm] fix transport fees for remote reserve transfers

### PR DESCRIPTION
**Pull request by girazoki**
Currently `transfer_assets` from pallet-xcm covers 4 main different transfer types:
- `localReserve`
- `DestinationReserve`
- `Teleport`
- `RemoteReserve`

For the first three, the local execution and the remote message sending are separated, and fees are deducted in pallet-xcm itself: https://github.com/paritytech/polkadot-sdk/blob/3410dfb3929462da88be2da813f121d8b1cf46b3/polkadot/xcm/pallet-xcm/src/lib.rs#L1758.

For the 4th case   `RemoteReserve`, pallet-xcm is still relying on the xcm-executor itself to send the message (through the `initiateReserveWithdraw` instruction). In this case, if delivery fees need to be charged, it is not possible to do so because the `jit_withdraw` mode has not being set.

This PR proposes to still use the `initiateReserveWithdraw` but prepending a `setFeesMode { jit_withdraw: true }` to make sure delivery fees can be paid.

A test-case is also added to present the aforementioned case

### CODE REVIEW

**Comment by acatangiu** [comment]
In the meantime, can you please add a prdoc for documenting the fix in the changelog?
Example(s) here: https://github.com/paritytech/polkadot-sdk/blob/master/prdoc/pr_3740.prdoc

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs [hunk 1]
```diff
@@ -2604,3 +2604,165 @@ fn teleport_assets_using_destination_reserve_fee_disallowed() {
 		expected_result,
 	);
 }
+
+/// Test `tested_call` with destination asset reserve and destination fee reserve using remote
+/// reserve.
```
Comment (acatangiu) [inline polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs:2609]:
```suggestion
/// Test `tested_call` transferring single asset using remote reserve.
```

File: polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs [hunk 2]
```diff
@@ -2604,3 +2604,165 @@ fn teleport_assets_using_destination_reserve_fee_disallowed() {
...
+}
+/// Test `transfer_assets` with remote asset reserve and remote fee reserve.
+#[test]
+fn transfer_assets_with_remote_asset_reserve_and_remote_asset_fee_reserve_paid_works() {
+	let expected_result = Ok(());
+	remote_asset_reserve_and_remote_fee_reserve_paid_call(
+		XcmPallet::transfer_assets,
+		expected_result,
+	);
+}
```
Comment (acatangiu) [inline polkadot/xcm/pallet-xcm/src/tests/assets_transfer.rs:2768]:
```suggestion
/// Test `transfer_assets` with remote asset reserve and remote fee reserve.
#[test]
fn transfer_assets_with_remote_asset_reserve_and_remote_asset_fee_reserve_paid_works() {
	let expected_result = Ok(());
	remote_asset_reserve_and_remote_fee_reserve_paid_call(
		XcmPallet::transfer_assets,
		expected_result,
	);
}

/// Test `limited_reserve_transfer_assets` with remote asset reserve and remote fee reserve.
#[test]
fn limited_reserve_transfer_assets_with_remote_asset_reserve_and_remote_asset_fee_reserve_paid_works() {
	let expected_result = Ok(());
	remote_asset_reserve_and_remote_fee_reserve_paid_call(
		XcmPallet::limited_reserve_transfer_assets,
		expected_result,
	);
}
```

**Comment by girazoki** [comment]
> In the meantime, can you please add a prdoc for documenting the fix in the changelog? Example(s) here: https://github.com/paritytech/polkadot-sdk/blob/master/prdoc/pr_3740.prdoc

Absolutely!

**Review by franciscoaguirre — APPROVED** [review]

**Comment by girazoki** [comment]
Merged @acatangiu

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5621324

**Comment by girazoki** [comment]
@acatangiu I dont know how to fix the prdoc failure now :(, do I need to put the title in between quotes?

**Review by acatangiu — APPROVED** [review]

**Pull request merged**